### PR TITLE
feat: Allow flexible SDK git repos

### PR DESCRIPTION
**Pull request by doosuu**
Allows more flexible configuration for SDK repository and subdirectories.

### CODE REVIEW

File: sdk-installer/src/run.py [hunk 1]
```diff
@@ -13,55 +13,130 @@
...
-        with open(requirements_path, encoding="utf-8") as requirements_file:
-            for line in requirements_file:
-                if line.startswith("velocitas_sdk") or line.startswith("velocitas-sdk"):
-                    sdk_version = line.split("==")[1].strip()
+    def is_package_installed(
+        self, package_name: str, package_version: Optional[str] = None
+    ) -> bool:
+        search_pattern = package_name
+        if package_version is not None:
+            search_pattern = f"{search_pattern}@{package_version}"
```
Comment (MP91) [inline sdk-installer/src/run.py:60]: Although technically correct I would prefer the following for better readability
```suggestion
            search_pattern = f"{package_name}@{package_version}"
```

File: sdk-installer/src/run.py [hunk 2]
```diff
@@ -13,55 +13,130 @@
...
+
+        output = subprocess.check_output(
+            ["conan", "search", package_name], encoding="utf-8"
```
Comment (MP91) [inline sdk-installer/src/run.py:63]:
```suggestion
            ["conan", "search", search_pattern], encoding="utf-8"
```

File: sdk-installer/src/run.py [hunk 3]
```diff
@@ -13,55 +13,130 @@
...
+        )
+        return output.find("Existing package recipes:") != -1
+
+    def get_required_package_version(self, package_name: str) -> str:
+        """Return the required version of the C++ SDK.
+
+        Returns:
+            str: The required version.
+        """
+        sdk_version: str = "0.4.1"
```
Comment (MP91) [inline sdk-installer/src/run.py:73]: can we read this default version from somewhere?

File: sdk-installer/src/run.py [hunk 4]
```diff
@@ -13,55 +13,130 @@
...
+
+        return package_found
+
+    def get_required_package_version(self, package_name: str) -> str:
+        """Return the required version of the Python SDK.
+
+        Returns:
+            str: The required version.
+        """
+        sdk_version: str = "0.13.0"
```
Comment (MP91) [inline sdk-installer/src/run.py:120]: same as above, can we get this default version from somewhere else? If not we should at least make it a global variable

File: sdk-installer/src/run.py [hunk 5]
```diff
@@ -13,55 +13,130 @@
...
+            package_data = line.strip().split(" ")
+            if package_data[0].startswith(package_name):
+                package_found = True
+                if package_version is not None:
+                    package_found = package_data[1] == package_version
+
+        return package_found
+
+    def get_required_package_version(self, package_name: str) -> str:
+        """Return the required version of the Python SDK.
```
Comment (MP91) [inline sdk-installer/src/run.py:115]: is this just for the SDK or for any package? either way we should adapt the name or the comment

File: sdk-installer/src/run.py [hunk 6]
```diff
@@ -80,69 +155,101 @@ def get_tag_or_branch_name(tag_or_branch_name: str) -> str:
...
-            "clone",
-            "--depth",
-            "1",
-            "-b",
-            get_tag_or_branch_name(required_sdk_version),
-            git_url,
-        ],
-        cwd=get_project_cache_dir(),
-    )
+    force_clone_repo(git_url, git_ref, package_clone_path, verbose_logging)
```
Comment (MP91) [inline sdk-installer/src/run.py:215]: will we also have a normal clone or something? if not we should rename this method to just clone_repo

**Review by MP91 — CHANGES_REQUESTED** [review]

File: sdk-installer/src/run.py (hunk 2 above)
Comment (doosuu) [reply sdk-installer/src/run.py:63]: Done

File: sdk-installer/src/run.py (hunk 5 above)
Comment (doosuu) [reply sdk-installer/src/run.py:115]: Comment adapted

File: sdk-installer/src/run.py (hunk 6 above)
Comment (doosuu) [reply sdk-installer/src/run.py:215]: Force clone will remove any previously created directory at the package_clone_path, hence the force.

File: sdk-installer/src/run.py (hunk 4 above)
Comment (doosuu) [reply sdk-installer/src/run.py:120]: Removed hardcoded version

File: sdk-installer/src/run.py (hunk 3 above)
Comment (doosuu) [reply sdk-installer/src/run.py:73]: Removed hardcoded version

File: setup/src/cpp/common/scripts/onCreateCommand.sh [hunk 7]
```diff
@@ -76,7 +76,7 @@ git submodule update --init
 echo "#######################################################"
 echo "### Install Dependencies                            ###"
 echo "#######################################################"
-velocitas exec build-system install
+./install_dependencies.sh 2>&1 | tee -a $HOME/install_dependencies.log
```
Comment (MP91) [inline setup/src/cpp/common/scripts/onCreateCommand.sh:79]: revert?

**Review by MP91 — CHANGES_REQUESTED** [review]

File: manifest.json [hunk 8]
```diff
@@ -103,7 +105,9 @@
                 {
                     "id": "run",
                     "executable": "python3",
-                    "args": [ "./conan-setup/src/run.py" ]
+                    "args": [
+                        "./components/conan-setup/src/run.py"
+                    ]
```
[the remark this answers is not in the document]
Comment (doosuu) [reply manifest.json:110]: Done

File: sdk-installer/src/run.py [hunk 9]
```diff
@@ -13,56 +13,134 @@
...
+    def get_required_package_version(self, package_name: str) -> Optional[str]:
+        """Return the required version of the given package from the
+        requirements file.
+
+        Returns:
+            str: The required version.
+        """
+        package_version: Optional[str] = None
+        with open(
+            os.path.join(get_workspace_dir(), "conanfile.txt"), encoding="utf-8"
```
[the remark this answers is not in the document]
Comment (doosuu) [reply sdk-installer/src/run.py:76]: Done

File: sdk-installer/src/run.py [hunk 10]
```diff
@@ -13,56 +13,134 @@
...
+        """
+        package_version: Optional[str] = None
+        requirements_path = os.path.join(
+            get_workspace_dir(), "app", "requirements-velocitas.txt"
+        )
+        if os.path.exists(requirements_path):
+            with open(requirements_path, encoding="utf-8") as requirements_file:
+                for line in requirements_file:
+                    if line.replace("-", "_").startswith(
+                        package_name.replace("-", "_")
```
[the remark this answers is not in the document]
Comment (doosuu) [reply sdk-installer/src/run.py:132]: It is done to be more lenient in package names we accept: An entry in conanfile `vehicle_app-sdk` could match package name `vehicle-app_sdk`

File: sdk-installer/src/run.py [hunk 11]
```diff
@@ -81,69 +159,118 @@ def get_tag_or_branch_name(tag_or_branch_name: str) -> str:
...
+    )
+
+    package_manager = get_package_manager(lang, verbose_logging)
+    required_package_version = package_manager.get_required_package_version(
+        package_dict["id"]
+    )
+
+    if required_package_version is None:
+        print(
+            f"No dependency on {package_dict['id']!r} detected -> Skipping installation."
```
[the remark this answers is not in the document]
Comment (doosuu) [reply sdk-installer/src/run.py:216]: If we are unable to find a required_package_version it means that there is no requirement towards vehicle-app-sdk. In which case we should not install the SDK.

File: setup/src/cpp/common/scripts/onCreateCommand.sh (hunk 7 above)
Comment (doosuu) [reply setup/src/cpp/common/scripts/onCreateCommand.sh:79]: Done

File: components/conan-setup/src/run.py
[the remark this answers is not in the document]
Comment (doosuu) [reply components/conan-setup/src/run.py:1]: Done

**Review by MP91 — APPROVED** [review]
lgtm

**Pull request merged**